tools/cache/oneshot-cache.go

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package cache
+
+import (
+	"fmt"
+	"sync"
+)
+
+// OneShotCache is a cache that calculates its value only once.
+// It will use the provided calculation function to get the value. Once a first call to this function results
+// in a non-error value, the cache will return this value for all subsequent calls.
+type OneShotCache[T any] struct {
+	value     T
```

Comment:
nit:  atomic.Value and you can drop `sync.Mutex`
nit: sync.Once instead of `isSet`

Reply:
`sync.Once` is problematic here, since I want this cache to support being called multiple times if the calculation function failed and returned an error. Under the hood, `sync.Once` uses a mutex, so the current implementation is very similar to it anyhow.

I did start using an `atomic.Pointer` though, following your suggestion! Thank you!